CMakeLists.txt

```diff
@@ -2,93 +2,181 @@ cmake_minimum_required (VERSION 3.5)
 
 project (korc_proj C CXX Fortran)
 
+#  FIXME: Remove once link issues are resolved with the new macos linker.
+if (${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
+   SET (CMAKE_EXE_LINKER_FLAGS -ld_classic)
+endif ()
+
+#-------------------------------------------------------------------------------
+#  Setup build types.
+#-------------------------------------------------------------------------------
+set_property (CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS
+              Debug
+              Release
+              Sanitized
+)
+
+#  Make sure the prefix path appears as a ccmake variable.
+set (CMAKE_PREFIX_PATH "" CACHE PATH "Search paths for dependency look up.")
+
+#-------------------------------------------------------------------------------
+#  Build Options
+#-------------------------------------------------------------------------------
+option (USE_ACC "Use OpenACC" OFF)
+option (USE_PSPLINE "Use PSPLINE interpolation" OFF)
+option (USE_FIO "Use FIO interpolation support" OFF)
+#FIXME: CORI is decomissioned. Is this necessary anymore?
+option (CORI_DIR "Use Cori directories" OFF)
+
+#-------------------------------------------------------------------------------
+#  Sanitizer options
+#-------------------------------------------------------------------------------
+add_library (sanitizer INTERFACE)
+target_compile_options (sanitizer
+                        INTERFACE
+                        $<$<CONFIG:Sanitized>:-g>
+)
+
+macro (register_sanitizer_option name default)
+    string (TOUPPER ${name} upper_name)
+    
+    option (SANITIZE_${upper_name} "Enable the ${name} sanitizer" ${default})
+
+    target_compile_options (sanitizer
+                            INTERFACE
+                            $<$<CONFIG:Sanitized>:$<$<BOOL:${SANITIZE_${upper_name}}>:-fsanitize=${name}>>
+    )
+    target_link_options (sanitizer
+                         INTERFACE
+                         $<$<CONFIG:Sanitized>:$<$<BOOL:${SANITIZE_${upper_name}}>:-fsanitize=${name}>>
+    )
+endmacro ()
+
+register_sanitizer_option (address ON)
```

Comment:
May we set all of these to `OFF` by default?

Reply:
We could but note these sanitizer options only have an effect when `CMAKE_BUILD_TYPE=Sanitized`. If we set all these to off that build type wouldn't do anything by default. This code was copied and pasted from a different project where I wanted an easy way to specify a build type in the GitHub actions strategy matrix. We could set these as Debug build only options with them all disabled by default.